docs/core/testing/unit-testing-with-copilot.md

```diff
@@ -0,0 +1,80 @@
+---
+title: Generate Unit Tests with Copilot
+author: sigmade
+description: How to generate unit tests and test projects in C# using the xUnit framework with the help of Visual Studio commands and GitHub Copilot
+ms.date: 01/12/2025
+---
+
+# Generate Unit Tests with GitHub Copilot and Visual Studio
+
+In this article, you explore how to generate unit tests and test projects in C# using the xUnit framework with the help of Visual Studio commands and GitHub Copilot.
+
+## Generate a test project and a stub method
+
+Imagine that there's a `ProductService` class with a `GetProductById` method that depends on the `IProductDataStorage` and `ICacheClient` interfaces.
+
+```csharp
+public class ProductService(
+    IProductDataStorage productDataStorage,
+    ICacheClient cacheClient)
+{
+    public async Task<Product?> GetProductById(int productId)
+    {
+        var product = await cacheClient.GetProduct(productId);
+
+        if (product is not null)
+        {
+            return product;
+        }
+
+        product = await productDataStorage.GetProduct(productId);
+
+        if (product is not null)
+        {
+            await _cacheClient.SetProduct(product);
+        }
+
+        return product;
+    }
+}
+```
+
+To generate a test project and a stub method, follow these steps:
+
+- Select the method.
+- Right-click and select **Create Unit Tests**.
+
+:::image type="content" source="media/create-unit-test.png" lightbox="media/create-unit-test.png" alt-text="Command Create Unit Tests":::
+
+In the **Create Unit Tests** dialog, select **xUnit** from the **Test Framework** dropdown menu.
+
+> [!NOTE]
+> The "Create Unit Tests" command defaults to the MSTest framework. However, since this example uses xUnit, you need to install the Visual Studio extension xUnit.net.TestGenerator2022.
```

Comment:
```suggestion
> The **Create Unit Tests** command defaults to the MSTest framework. However, since this example uses xUnit, you need to install the Visual Studio extension [xUnit.net.TestGenerator2022](https://marketplace.visualstudio.com/items?itemName=YowkoTsai.xunitnettestgenerator2022).
```